libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteProgressEx.kt

```diff
@@ -8,35 +8,61 @@ import com.mapbox.api.directions.v5.models.RouteOptions
 import com.mapbox.geojson.Point
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.base.utils.DecodeUtils.stepGeometryToPoints
+import com.mapbox.navigation.utils.internal.ifNonNull
 import com.mapbox.navigation.utils.internal.logD
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 private const val LOG_CATEGORY = "NavigationRouteUtils"
 
 /**
- * This function checks whether the [NavigationRoute] has unexpected closures, which could be a reason to re-route.
+ * This function checks whether the [NavigationRoute] has unexpected upcoming closures, which could be a reason to re-route.
```

Comment:
what is the difference between unexpected and expected closure? Is the unexpected the one we receive after route refresh?

Reply:
the difference is in snapping options (see [com.mapbox.api.directions.v5.models.RouteOptions](https://github.com/mapbox/mapbox-java/blob/9c95f5713f95dbcccfbc8f787f3b519e311b4f9a/services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java#L740-L775)). 

Whenever snapping is not allowed, that means the closure is unexpected